cli/internal/packagemanager/pnpm.go

```diff
@@ -119,4 +120,49 @@ var nodejsPnpm = PackageManager{
 	readLockfile: func(contents []byte) (lockfile.Lockfile, error) {
 		return lockfile.DecodePnpmLockfile(contents)
 	},
+
+	prunePatches: func(pkgJSON *fs.PackageJSON, patches []turbopath.AnchoredUnixPath) error {
+		return pnpmPrunePatches(pkgJSON, patches)
+	},
+}
+
+func pnpmPrunePatches(pkgJSON *fs.PackageJSON, patches []turbopath.AnchoredUnixPath) error {
+	pkgJSON.Mu.Lock()
+	defer pkgJSON.Mu.Unlock()
+
+	keysToDelete := []string{}
+	pnpmConfig, ok := pkgJSON.RawJSON["pnpm"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("Invalid structure for pnpm field in package.json")
+	}
+	patchedDependencies, ok := pnpmConfig["patchedDependencies"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("Invalid structure for patchedDependencies field in package.json")
+	}
+
+	for dependency, patch := range patchedDependencies {
+		patch, ok := patch.(string)
+		if !ok {
+			return fmt.Errorf("Expected only strings in patchedDependencies. Got %v", patch)
```

Comment:
Are we guaranteed to have a usable value of `patch` in this error? I think the failed cast is also going to reassign `patch`